wolframalpha/__init__.py

```diff
@@ -1,84 +1,184 @@
+    """
+    def __init__(self, app_id='Q59EW4-7K8AHE858R'):
+        self.app_id = app_id
+
+    def query(self, query, assumption=None):
+        """
+        Query Wolfram|Alpha using the v2.0 API
+        Allows for assumptions to be included.
+        See: http://products.wolframalpha.com/api/documentation.html#6
+        """
+        data = {
+                'input': query,
+                'appid': self.app_id
+        }
+        if assumption:
+            data.update({'assumption': assumption})
+        
+        query = urllib.parse.urlencode(data)
+        url = 'https://api.wolframalpha.com/v2/query?' + query
+        resp = urllib.request.urlopen(url)
+        assert resp.headers.get_content_type() == 'text/xml'
+        assert resp.headers.get_param('charset') == 'utf-8'
+        return Result(resp)
+
 class Result(object):
+    '''
+    Handles processing the response for the programmer.
+    '''
     def __init__(self, stream):
-        self.tree = etree.parse(stream)
+        self.tree = xmltodict.parse(stream, dict_constructor=dict)['queryresult']
         self._handle_error()
-
+        self.info = []
+        try:
+            self.pods = list(map(Pod, self.tree['pod']))
+            self.info.append(self.pods)
+        except KeyError:
+            self.pods = None
+        try:
+            self.assumptions = list(map(Assumption, self.tree['assumptions']))
+            self.info.append(self.assumptions)
+        except KeyError:
+            self.assumptions = None
+        try:
+            self.warnings = list(map(Warning, self.tree['warnings']))
+            self.info.append(self.warnings)
+        except KeyError:
+            self.warnings = None
+    
     def _handle_error(self):
-        error = self.tree.find('error')
-        if not error:
+        error_state = self.tree['@error']
+        if error_state == 'false':
             return
-
-        code = error.find('code').text
-        msg = error.find('msg').text
-        tmpl = 'Error {code}: {msg}'
-        raise Exception(tmpl.format(code=code, msg=msg))
-
+        
+        error = self.tree['error']
+        code = error['code']
+        msg = error['msg']
+        template = 'Error {code}: {msg}'
+        raise Exception(template.format(code=code, msg=msg))
+    
     def __iter__(self):
-        return (Pod(node) for node in self.tree.findall('pod'))
-
+        return iter(self.info)
+    
     def __len__(self):
-        return len(self.tree)
-
-    @property
-    def pods(self):
-        return list(iter(self))
-
+        return len(self.info)
+    
     @property
     def results(self):
-        return (pod for pod in self if pod.title=='Result')
+        ''' Get the pods that hold the response to a simple, discrete query. '''
+        return [pod for pod in self.pods if pod.primary or pod.title=='Result']
+    
+    @property
+    def details(self):
+        ''' Get a simplified set of answers with some context. '''
```

Comment:
For docstrings, let's stick with the convention already established in the file and use triple-double quotes and newlines instead of spaces to separate the quotes from the text.